src/components/ReportModals/CashDrawerReportModal.js

```diff
@@ -65,24 +65,20 @@ const CashDrawerReportModal = (props) => {
     value: id,
     label: name,
   }));
-  const getSources = async (value) => {
-    if (!value) {
-      return undefined;
-    }
-
-    const response = await cashDrawerReportSources.POST({ createdAt: value });
-    const responseSources = response?.sources ?? [];
-    const formatSources = responseSources.map((sourceName, index) => ({
-      value: `source-${index}`,
-      label: sourceName,
-    }));
-    setSources(formatSources || []);
 
-    return undefined;
-  };
-
-  useMemo(() => getSources(servicePointsValue), [servicePointsValue])
-    .then(response => response);
+  useMemo(() => {
```

Comment:
In this case, I would expect to see `useEffect` then.  `useMemo` is more for storing the value which will be passed somewhere later (e.g. as props to child component)